mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/handler/MqttSubscribeHandler.java

```diff
@@ -128,4 +160,66 @@ private MqttSubAckMessage getResponse(MqttSubscribeMessage mqttSubscribeMessage)
         return mqttSubAckMessage;
     }
 
+
+    @SuppressWarnings("checkstyle:Indentation")
+    private void sendRetainMessage(ChannelHandlerContext ctx, Set<Subscription> subscriptions) throws InterruptedException, RemotingException, org.apache.rocketmq.remoting.exception.RemotingException {
+
+        String clientId = ChannelInfo.getClientId(ctx.channel());
+        Session session = sessionLoop.getSession(ChannelInfo.getId(ctx.channel()));
+        Set<Subscription> preciseTopics = new HashSet<>();
+        Set<Subscription> wildcardTopics = new HashSet<>();
+
+        for (Subscription subscription : subscriptions) {
+            if (!TopicUtils.isWildCard(subscription.getTopicFilter())) {
+                preciseTopics.add(subscription);
+            } else {
+                wildcardTopics.add(subscription);
+            }
+        }
+
+        for (Subscription subscription : preciseTopics) {
+            CompletableFuture<Message> retainedMessage = retainedPersistManager.getRetainedMessage(subscription.getTopicFilter());
+            retainedMessage.whenComplete((msg, throwable) -> {
+                if (msg == null) {
+                    return;
+                }
+                _sendMessage(session, clientId, subscription, msg);
+            });
+        }
+
+        for (Subscription subscription : wildcardTopics) {
+
+            CompletableFuture<ArrayList<String>> future = retainedPersistManager.getMsgsFromTrie(subscription);
+            future.whenComplete((msgsList,throwable) -> {
+                ArrayList<Message> results = new ArrayList<>();
+                for (String strMsg:msgsList) {
+                    results.add(JSON.parseObject(strMsg,Message.class));
```

Comment:
results.add(JSON.parseObject(strMsg,Message.class));    Can it be optimized here？